### PR TITLE
Fix edit rotation handle fidgeting

### PR DESCRIPTION
**Pull request by huffman**
https://highfidelity.fogbugz.com/f/cases/17471/Create-Fix-rotation-tool-fidgeting-around-while-rotating

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13107/HighFidelity-Beta-PR14026-8eec03a.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13114/HighFidelity-Beta-PR14026-431daec.apk

**Comment by huffman** [comment]
# Test Case

 1. https://highfidelity.testrail.net/index.php?/cases/view/17771
 2. Ensure that the rotation handles do not fidget when rotating on any of the 3 axes. An example of what shouldn't happen is: 
edit-rotate-fidget

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13114/HighFidelity-Beta-PR14026-431daec.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13114/HighFidelity-Beta-Interface-PR14026-431daec.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13114/HighFidelity-Beta-PR14026-431daec.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13114/HighFidelity-Beta-Interface-PR14026-431daec.exe

**Comment by qliacoombs** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13215/HighFidelity-Beta-PR14026-431daec.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13215/HighFidelity-Beta-PR14026-431daec.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13215/HighFidelity-Beta-Interface-PR14026-431daec.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13215/HighFidelity-Beta-PR14026-431daec.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13215/HighFidelity-Beta-Interface-PR14026-431daec.exe

**Comment by jakram** [comment]
Currently testing

**Comment by qlihifitest** [comment]
Currently testing

**Comment by qlihifitest** [comment]
PASS @Conklin94122

**Comment by qlihifitest** [comment]
PASS @Conklin94122 

Now works as intended, slider no longer fidgets when used

File: interface/src/ui/overlays/Circle3DOverlay.cpp [hunk 1]
```diff
@@ -111,28 +111,38 @@ void Circle3DOverlay::render(RenderArgs* args) {
...
+
+            const auto totalRange = _endAt - _startAt;
             if (_innerRadius <= 0) {
                 _solidPrimitive = gpu::TRIANGLE_FAN;
                 points << vec2();
                 colors << innerStartColor;
-                for (float angle = _startAt; angle <= _endAt; angle += SLICE_ANGLE) {
-                    float range = (angle - _startAt) / (_endAt - _startAt);
-                    float angleRadians = glm::radians(angle);
+                for (float angleRadians = startAtRadians; angleRadians < endAtRadians; angleRadians += SLICE_ANGLE_RADIANS) {
```
Comment (dback2) [inline interface/src/ui/overlays/Circle3DOverlay.cpp]: We don't want angleRadians <= endAtRadians? Or was that part of the issue before

File: scripts/system/libraries/entitySelectionTool.js [hunk 2]
```diff
@@ -2622,6 +2622,10 @@ SelectionDisplay = (function() {
                     updateSelectionsRotation(rotationChange, rotationCenter);
                     updateRotationDegreesOverlay(-angleFromZero, rotationDegreesPosition);
 
+                    if (direction === ROTATE_DIRECTION.YAW) {
+                        angleFromZero *= -1;
```
Comment (dback2) [inline scripts/system/libraries/entitySelectionTool.js]: Looks a little better than my case for yaw below hehe ;)

File: interface/src/ui/overlays/Circle3DOverlay.cpp (hunk 1 above)
Comment (huffman) [reply interface/src/ui/overlays/Circle3DOverlay.cpp]: That was related. The semicircles are drawn in sections of 2 degrees, but if a semicircle was not a multiple of 2, the final section (say, a 1.2 degree section on a 11.2 degree semicircle) would not be drawn. We now draw that final section, which might be the last 2 degree section. If we did `<=` we would draw the last section twice and then draw a 0 degree section.

File: scripts/system/libraries/entitySelectionTool.js [hunk 3]
```diff
@@ -2635,16 +2639,6 @@ SelectionDisplay = (function() {
                     });
 
                     // not sure why but this seems to be needed to fix an reverse rotation for yaw ring only
```
Comment (dback2) [inline scripts/system/libraries/entitySelectionTool.js]: Sorry for late comment but just noticed this comment should be removed too

**Comment by huffman** [comment]
@dback2 Updated

No QA pass needed, only change is a removed comment.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13476/HighFidelity-Beta-PR14026-feb230b.apk

**Comment by conklin94122** [comment]
Dev changes have triggered a rebuild. QA will retest, labeled as QA Ready.

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13476/HighFidelity-Beta-PR14026-feb230b.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13476/HighFidelity-Beta-Interface-PR14026-feb230b.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13476/HighFidelity-Beta-PR14026-feb230b.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13476/HighFidelity-Beta-Interface-PR14026-feb230b.dmg

**Pull request merged**